llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -1910,20 +1915,12 @@ bool X86DAGToDAGISel::matchAddress(SDValue N, X86ISelAddressMode &AM) {
 
   // Post-processing: Convert foo to foo(%rip), even in non-PIC mode,
   // because it has a smaller encoding.
```

Comment:
Aha, this is probably doing the most work to fix small / medium code model discrepancies.